t/t2801-top-cmd.t

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+test_description='Test flux top command'
+
+. $(dirname $0)/sharness.sh
+
+test_under_flux 4
+
+runpty="${SHARNESS_TEST_SRCDIR}/scripts/runpty.py"
+
+test_expect_success 'flux-top -h prints custom usage' '
+	flux top -h 2>usage &&
+	grep "Usage:.*JOBID" usage
+'
+test_expect_success 'flux-top fails on unknown option' '
+	test_must_fail flux top --notopt 2>notopt.err &&
+	grep "unrecognized option" notopt.err
+'
+test_expect_success 'flux-top fails if FLUX_URI is set wrong' '
+	FLUX_URI=noturi test_must_fail flux top 2>baduri.err &&
+	grep "connecting to Flux" baduri.err
+'
+test_expect_success 'flux-top fails if job argument is not a job ID' '
+	test_must_fail flux top notjob 2>badjobid.err &&
+	grep "failed to parse JOBID" badjobid.err
+'
+test_expect_success 'flux-top fails if job argument is unknown' '
+	test_must_fail flux top 12345 2>unkjobid.err &&
+	grep "unknown job" unkjobid.err
+'
+test_expect_success 'run a test job to completion' '
+	flux mini submit -n1 /bin/true >jobid &&
+	flux job attach $(cat jobid)
+'
+test_expect_success 'flux-top fails if job is not running' '
+	test_must_fail flux top $(cat jobid) 2>notrun.err &&
+	grep "job is not running" notrun.err
+'
+test_expect_success 'flux-top fails if stdin is not a tty' '
+	test_must_fail flux top --test-exit </dev/null 2>notty.err &&
+	grep "stdin is not a terminal" notty.err
+'
+test_expect_success 'flux-top --test-exit works with a pty' '
+	$runpty flux top --test-exit >/dev/null
```

Comment:
Note that the `runpty` script already supports a "quit character" sent when the script receives `SIGUSR1`.
Instead of adding a test-only `--test-exit` option, you could use something like the following:
```sh
$runpty --format=asciicast -o log.asciicast --quit-char=q src/cmd/flux top &
pid=$! &&
kill -USR1 $pid &&
wait $pid
```

This could be a more flexible way to test `flux top` since you can wait for an event to occur before quitting the program (and it actually tests using `q` to quit the program), and additionally the screen cast is captured in the asciicast v2 output so you can actually check for (some kinds of) expected output (though with ncurses that could be challenging).

It would also be feasible to eventually extend `runpty.py` to add a scripted list of input if we want to ever add more testing based on keystrokes.